features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/Debouncer.kt

```diff
@@ -29,10 +32,12 @@ internal class Debouncer(
         }
         handler.removeCallbacksAndMessages(null)
         val timePassedSinceLastExecution = System.nanoTime() - lastTimeRecordWasPerformed
-        if (timePassedSinceLastExecution >= maxRecordDelayInNs) {
-            executeRunnable(runnable)
-        } else {
-            handler.postDelayed({ executeRunnable(runnable) }, DEBOUNCE_TIME_IN_MS)
+        runInTimeBalance {
```

Comment:
is this needed to handle the problem in Android 23 and below where the SR debouncer doesn't work ?

Reply:
The task is not aimed to handle the infinite calling of recorder problem under Android 23, it's an overall optimisation of all versions, but I believe it can somehow solve it by the way, because if under Android 23, the snapshot is recorded too frequently, it will very soon use out the time balance in the bank, and following recording will be refused due to this optimisation.